### PR TITLE
feat: Log when error occurs while pushing

### PR DESCRIPTION
**Pull request by jinwoo**
And upgrade `h2-auto-push` to v0.4.0.

Also update other packages.

### CODE REVIEW

File: ts/src/index.ts [hunk 1]
```diff
@@ -70,7 +70,10 @@ async function staticServeFn(
           'set-cookie',
           cookie.serialize(CACHE_COOKIE_KEY, newCacheCookie, {path: '/'}));
 
-      pushFn(reqStream).then(noop, noop);
+      reqStream.on('pushError', (err) => {
+        app.log.error('Error while pushing', err);
```
Comment (delvedor) [inline ts/src/index.ts]: You can use `req.log.error`, in this way the log will also have the request id and support the [custom log leve](https://github.com/fastify/fastify/blob/master/docs/Routes.md#custom-log-level)l on plugin/route basis :)

Comment (jinwoo) [reply ts/src/index.ts]: I wanted to, but here in the `onRequest` hook, the request object is not a fastify request but a raw request. Do I have a better option?

Comment (delvedor) [reply ts/src/index.ts]: `log` is present in both [fastify request](https://github.com/fastify/fastify/blob/master/lib/handleRequest.js#L14) and [raw request](https://github.com/fastify/fastify/blob/master/fastify.js#L204) :)

Comment (jinwoo) [reply ts/src/index.ts]: Ah, good to know. Thanks. Updated.

**Pull request merged**